benchmark-service/benchmark/benchmark.py

```diff
@@ -45,18 +47,74 @@ def get_benchmarks(file_path):
             stats['stdev'] = round(d['stdev'], 6)
             stats['times'] = d['times']
             stats['trials'] = d['trials']
-        data.append(stats)
+        data[stats['name']] = stats
     geometric_mean = get_geometric_mean(prod_of_means, len(pre_data['benchmarks']))
 
     file_info = parse_file_path(BENCHMARK_FILE_REGEX, file_path)
     sha = file_info['sha']
     benchmarks = dict()
     benchmarks['sha'] = sha
     benchmarks['geometric_mean'] = geometric_mean
-    benchmarks['data'] = sorted(data, key=lambda i: i['name'])
+    benchmarks['data'] = data
     return benchmarks
 
 
+def get_comparisons(benchmarks1, benchmarks2, metric):
+    def get_metric(data):
+        if metric == 'median':
+            return data['median']
+        assert metric == 'best'
+        return min(data['times'])
+
+    set_of_names = set()
+    for name1 in benchmarks1['data'].keys():
+        set_of_names.add(name1)
+    for name2 in benchmarks2['data'].keys():
+        set_of_names.add(name2)
+
+    comparisons = []
+    for name in set_of_names:
+        data1 = benchmarks1['data'].get(name)
+        data2 = benchmarks2['data'].get(name)
+        if data2 is None:
+            comparisons.append((name, get_metric(data1), None))
+        elif data1 is None:
+            comparisons.append((name, None, get_metric(data2)))
+        else:
+            comparisons.append((name, get_metric(data1), get_metric(data2)))
+
+    return comparisons
+
+
+def fmt_time(t):
+    return round(t, 3)
+
+
+def fmt_diff(ratio):
+    return round(ratio * 100, 3)
+
+
+def final_comparisons(comparisons):
+    comps = []
+    ratios = []
+    final_comps = {}
+    for name, r1, r2 in comparisons:
+        if r1 is None:
+            comps.append((name, None, None, fmt_time(r2)))
+        elif r2 is None:
+            comps.append((name, None, fmt_time(r1), None))
+        else:
+            r = r1 / r2
+            ratios.append(r)
+            comps.append((name, fmt_diff(r), fmt_time(r1), fmt_time(r2)))
+    final_comps['comps'] = comps
+    final_comps['harmonic_mean'] = fmt_diff(hmean(ratios))
```

Comment:
Can we do something here where if the ratios is empty, then we set these values to None? Then in the Jinja template, we only show the value if it's not None.